Library/ViewModels/CommentsViewModelTests.swift

```diff
@@ -111,365 +113,365 @@ internal final class CommentsViewModelTests: TestCase {
+      self.vm.inputs.viewDidLoad()
 
-    func testProjectPagination_WhenLimitReached_CommentsAreUpdatedInDataSource() {
       self.loadCommentsAndProjectIntoDataSourceComments.assertDidNotEmitValue()
       self.loadCommentsAndProjectIntoDataSourceProject.assertDidNotEmitValue()
+      self.isCommentsLoading.assertValues([true])
 
-      let envelope = CommentsEnvelope.singleCommentTemplate
-      let project = Project.template
+      self.scheduler.advance()
 
-      withEnvironment(apiService: MockService(fetchCommentsEnvelopeResult: .success(envelope))) {
-        self.vm.inputs.configureWith(project: project, update: nil)
-        self.vm.inputs.viewDidLoad()
+      self.loadCommentsAndProjectIntoDataSourceComments.assertValues(
+        [envelope.comments],
+        "A set of comments is emitted."
+      )
+      self.loadCommentsAndProjectIntoDataSourceProject.assertValues([.template])
+      self.isCommentsLoading.assertValues([true, false])
 
-        self.loadCommentsAndProjectIntoDataSourceComments.assertDidNotEmitValue()
-        self.loadCommentsAndProjectIntoDataSourceProject.assertDidNotEmitValue()
-        self.isCommentsLoading.assertValues([true])
+      let updatedEnvelope = CommentsEnvelope.multipleCommentTemplate
 
-        self.scheduler.advance()
+      withEnvironment(apiService: MockService(fetchCommentsEnvelopeResult: .success(updatedEnvelope))) {
+        self.vm.inputs.willDisplayRow(3, outOf: 4)
 
         self.loadCommentsAndProjectIntoDataSourceComments.assertValues(
           [envelope.comments],
-          "A set of comments is emitted."
+          "No new comments are emitted."
         )
+
         self.loadCommentsAndProjectIntoDataSourceProject.assertValues([.template])
-        self.isCommentsLoading.assertValues([true, false])
+        self.isCommentsLoading.assertValues([true, false, true])
+        self.shouldShowIndicator.assertValues([false, true])
 
-        let updatedEnvelope = CommentsEnvelope.multipleCommentTemplate
+        self.scheduler.advance()
 
-        withEnvironment(apiService: MockService(fetchCommentsEnvelopeResult: .success(updatedEnvelope))) {
-          self.vm.inputs.willDisplayRow(3, outOf: 4)
+        self.loadCommentsAndProjectIntoDataSourceComments.assertValueCount(2)
 
-          self.loadCommentsAndProjectIntoDataSourceComments.assertValues(
-            [envelope.comments],
-            "No new comments are emitted."
-          )
-          self.loadCommentsAndProjectIntoDataSourceProject.assertValues([.template])
-          self.isCommentsLoading.assertValues([true, false, true])
+        self.loadCommentsAndProjectIntoDataSourceComments.assertValues(
+          [envelope.comments, envelope.comments + updatedEnvelope.comments],
+          "New comments are emitted."
+        )
+        self.loadCommentsAndProjectIntoDataSourceProject.assertValues([.template, .template])
+        self.isCommentsLoading.assertValues([true, false, true, false])
```

Comment:
test `shouldShowIndicator` here as well.

Reply:
As you've mentioned in one of the comments, rename would do well, testing `shouldShowIndicator` would return empty at those stages you mentioned.